Single source onboarding documentation (#119)
mozilla/build-relengdocs: procedures/accounts_setup.md

Also don't forget to add the fingerprint under https://login.mozilla.com/ too. Moreover, you'll have to add your GPG key to Github and Git. This is usuaully optional but is a requirement in some repositories. Follow [these instructions to add your key to Github](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account) and [these instructions to register your key with Git](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key) for more details.
* VPN - Many of our systems are behind a private network in addition to auth0. Follow the prompts to generate and download an openVPN certificate that you can use to import to your vpn client. See the instructions on how to [install and configure your VPN client](https://mana.mozilla.org/wiki/display/SD/VPN) and help choosing the right client for your platform. Note: macOS and Windows users should use [Viscosity](https://www.sparklabs.com/viscosity/). This application comes with a free 30 day trial. During your trial, your manager can help you create a JIRA ticket to get a Viscosity full license.

> * MFA - This MFA account is specific to https://login.mozilla.com/duo/ is used for LDAP/auth0 based logins. Follow the instructions to download the Duo Mobile app and create a Mozilla account

Reviewer: First sentence is awkward here.

Author: Good catch, I've added the correct mana URL and addressed this, thank you.

* Mercurial - Most development in releng (and at Mozilla writ-large) is stored in version control using [hg](http://mercurial.selenic.com/). There is an excellent step-by-step guide for setting up and using hg: [Mercurial for Mozillians](https://mozilla-version-control-tools.readthedocs.org/en/latest/hgmozilla/index.html).
* Phabricator - you'll be granted access to this via SSO
The root webview of the Mozilla hg repositories is here: https://hg.mozilla.org/ while most releng code lives in repos under https://hg.mozilla.org/build.

> Please fill out the [Commit Access form](https://www.mozilla.org/en-US/about/governance/policies/commit/access-policy/) and submit it via a bug on file such as [this](TODO help from Heitor here).

Author: @hneiva help with this section :)

Reviewer 2: ```suggestion
Please fill out the [Commit Access form](https://www.mozilla.org/en-US/about/governance/policies/commit/access-policy/) and submit it via a bug on file such as [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1734247). Make sure to include (comment) your SSH public key and request 2 peers to vouch for you.
```

### Calendar

Like mail, we now use [Google calendar](https://www.google.com/calendar/)

You'll want to subscribe to the following public calendars:
* [RelEng MoCo calendar](https://calendar.google.com/calendar/u/0?cid=bW96aWxsYS5jb21fMmQzMjM0MzMzMzM1MzAzNjMxMmQzOTM3MzdAcmVzb3VyY2UuY2FsZW5kYXIuZ29vZ2xlLmNvbQ)
* [RelEng PTO calendar](https://calendar.google.com/calendar/u/0?cid=Y190cWEzaHQ5bXB0ODN1djJzZDJsYjVsaTVva0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t)

Talk to your manager/mentor to get added to the various other private calendars as appropriate.

### Matrix

The majority of day-to-day communication in releng happens on Matrix. More details on this can be found on [wiki Matrix](https://wiki.mozilla.org/Matrix).
Please talk to your manager about the private/public channels you should be in.

### Slack

Some parts of Mozilla prefer Slack to Matrix, more info on [mana](https://mana.mozilla.org/wiki/display/SD/Slack).

### Zoom

For video communication we use Zoom. More details can be found on [mana](https://mana.mozilla.org/wiki/display/AVSE/Getting+Started+With+Zoom)

### Wiki

Historical pieces of Mozilla information can be found in wiki.

> TODO - ask Heitor for instructions on how to get wiki.m.o account

Author: @hneiva help with this section :)

Reviewer 2: ```suggestion
Request an account filling out [this form](https://wiki.mozilla.org/Special:RequestAccount). You should receive an email with a password in 1-2 days. If not, [file a bug in bugzilla](https://bugzilla.mozilla.org/enter_bug.cgi?product=Websites&component=wiki.mozilla.org)
```

### Google Drive

Google Drive (formerly Google docs) is a preferred way to share things these days. This includes spreadsheets and documents that will change a great deal over time.
Google Drive access should be enabled with your email account when you start. If you need access to a particular document, talk to the document owner or your manager/mentor.

## Bugzilla

Almost everything at Mozilla goes through Bugzilla. [Create a Bugzilla account](https://bugzilla.mozilla.org/createaccount.cgi) if you have not already.
You'll need a few tweaks to your account to get access to everything releng-related:
* Add privileges for bugzilla group releng confidential (Can be done by manager or bugzilla admin)
* Add your comms nickname & ldap username as "aliases" for your account
* log into bugzilla & follow links "Preferences" -> "Account Information"
* append the aliases, with a leading ':' and enclosed in brackets ('[]') to the "Real Name" field
* e.g.: "John Doe [:jdoe]"

The product to use is "Release Engineering." There are multiple possible components under that product, so take your best guess or ask for guidance from the team.

> TODO - to revisit once we reorg Bugzilla in November 2022.

Author: Note to self.

Reviewer 2: I feel like Bugzilla should be mentioned first, as some of the above accounts require filing a bug in it.

Author: Good catch, moved at the top.

* Ship-it - follow procedure in [here](https://moz-releng-docs.readthedocs.io/en/latest/procedures/release-duty/index.html#how-to-get-ship-it-access) to get access.
* Balrog - follow procedure in [here](https://moz-releng-docs.readthedocs.io/en/latest/procedures/release-duty/index.html#how-to-get-balrog-access) to get access. A similar related bug was https://bugzilla.mozilla.org/show_bug.cgi?id=1727341.
* CultureAmp - talk to your manager to set your goals and run the logistics in there for the 1x1 tracking. You'll also need to be added to the RelEng team goals.
* JIRA - access to JIRA is granted upon SSO access so please make sure to login via the main dashboard. Ask your manager to add you to the RELENG/FFXP boards.
* Sentry logs - File a bug similar to [this one](https://bugzilla.mozilla.org/show_bug.cgi?id=1731311) to get access to debug various logs in Balrog and more.
* CloudOps Jenkins - File a bug similar to [Bug 1721444](https://bugzilla.mozilla.org/show_bug.cgi?id=1721444) and talk to your manager to get access to Janekins CloudOps to be able to debug. You should already have access to cloudops-infra repo if you've done the Github section above.

> ## Jumphosts

Author: @bhearsum I'm curious, do we still need this? I vaguely recall we retired some of the jumphosts.

* CultureAmp - talk to your manager to set your goals and run the logistics in there for the 1x1 tracking. You'll also need to be added to the RelEng team goals.
* JIRA - access to JIRA is granted upon SSO access so please make sure to login via the main dashboard. Ask your manager to add you to the RELENG/FFXP boards.
* Sentry logs - File a bug similar to [this one](https://bugzilla.mozilla.org/show_bug.cgi?id=1731311) to get access to debug various logs in Balrog and more.
* CloudOps Jenkins - File a bug similar to [Bug 1721444](https://bugzilla.mozilla.org/show_bug.cgi?id=1721444) and talk to your manager to get access to Janekins CloudOps to be able to debug. You should already have access to cloudops-infra repo if you've done the Github section above.
Once it's properly configured, you'll be able to access it by making some tweaks documented [here](https://github.com/mozilla-services/cloudops-deployment#accessing-jenkins).

### LDAP groups

> * RelEng LDAP flags to access various infrastructure pieces. File a bug similar to [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1732846) and ask your manager for vouching. The flags needed are listed below:

Author: @gbrownmozilla I used your bug for reference. I'm curious, did you need any other flags to be enabled in the past three months?

Reviewer: Oh, yes, actually! I needed "vpn_cloudops_shipit" (bug 1727334).

Author: Great catch, added.

## Accounts setup
Here is a list of accounts and access needed to be successful in Release Engineering.

* SSO - we rely on [auth0](https://auth0.com/) across Mozilla for authentication and [LDAP](https://mana.mozilla.org/wiki/display/SYSADMIN/LDAP+Architecture) for authorization.
Once given LDAP and you have created a permanent password, you can use that to login to the [portal](https://sso.mozilla.com/). From SSO, you should have links to various services from email, calendar, slack, mana, etc.
* https://login.mozilla.com/ is where you can change a number of authentication/authorization access bits that you have control over. Each todo in this section assumes you have access to this page.
* SSH - upload your public ssh key. It is a good idea to generate a separate ssh keypair from your personal one or any other that you have created in the past and use that explicitly for Releng and upload that. Follow this [SSH guidelines](https://wiki.mozilla.org/Security/Guidelines/OpenSSH#OpenSSH_client) doc on how to generate, configure, and use your ssh key.

> * PGP - We use pgp keys to share private information, secrets, and verify that the source came from someone we trust. Generate a keypair for this and upload your public key to https://keys.openpgp.org/ so others can find it. It would be really good if you could have other people sign your key, adding more trust that this key really belongs to you. You can use the the [pgp quickstart](https://mana.mozilla.org/wiki/display/SD/Generating+a+GPG+Public+Key) guide on mana or you can use the The [GNU Privacy Handbook](https://www.gnupg.org/gph/en/manual.html) for reference.

Reviewer 2: ```suggestion
* GPG - We use gpg keys to share private information, secrets, and verify that the source came from someone we trust. Generate a keypair and upload your public key to [OpenPGP](https://keys.openpgp.org/) so others can find it. It would be really good if you could have other people sign your key, adding more trust that this key really belongs to you. You can use the the [GPG quickstart](https://mana.mozilla.org/wiki/display/SD/Generating+a+GPG+Public+Key) guide on mana or you can use the The [GNU Privacy Handbook](https://www.gnupg.org/gph/en/manual.html) for reference.
```

GPG, PGP... so confusing... PGP is owned by Symantec (commercial license), OpenPGP is an email-centric service (but also used for identity trust services), and GPG (aka GnuPG) is the system/key format we use for signing commits/messages/data. I don't know why they made the names so similar. AFAIK GnuPG was based on PGP. Anyway... \<tableflip\>

Also don't forget to add the fingerprint under https://login.mozilla.com/ too. Moreover, you'll have to add your GPG key to Github and Git. This is usuaully optional but is a requirement in some repositories. Follow [these instructions to add your key to Github](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account) and [these instructions to register your key with Git](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key) for more details.
* VPN - Many of our systems are behind a private network in addition to auth0. Follow the prompts to generate and download an openVPN certificate that you can use to import to your vpn client. See the instructions on how to [install and configure your VPN client](https://mana.mozilla.org/wiki/display/SD/VPN) and help choosing the right client for your platform. Note: macOS and Windows users should use [Viscosity](https://www.sparklabs.com/viscosity/). This application comes with a free 30 day trial. During your trial, your manager can help you create a JIRA ticket to get a Viscosity full license.

> * MFA - More information on Mutliple-Factor-Authentication can be found in https://login.mozilla.com/duo/ which is what we use for LDAP/auth0 based logins. Follow [these](https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=145590267) instructions to download the Duo Mobile app and set it up.

Reviewer 2: ```suggestion
* MFA - More information on Multi-Factor-Authentication can be found in the [Login Portal](https://login.mozilla.com/duo/) which is what we use for LDAP/auth0 based logins. Follow [these](https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=145590267) instructions to download the Duo Mobile app and set it up.
```

* Phabricator - you'll be granted access to this via SSO
The root webview of the Mozilla hg repositories is here: https://hg.mozilla.org/ while most releng code lives in repos under https://hg.mozilla.org/build.
Please fill out the [Commit Access form](https://www.mozilla.org/en-US/about/governance/policies/commit/access-policy/) and submit it via a bug on file such as [this](TODO help from Heitor here).

## Git & Github
Please file a bug similar to [bug 1726348](https://bugzilla.mozilla.org/show_bug.cgi?id=1726348) and include the following teams to be added. Ask for your manager to vouch in the bug for these changes. The teams you need to be added are:
1. https://github.com/orgs/mozilla/teams/releng/members
2. https://github.com/orgs/mozilla-releng/people
3. https://github.com/orgs/mozilla-releng/teams/releng/members
4. https://github.com/orgs/mozilla-mobile/teams/releng/members

> 5. https://github.com/mozilla-services/cloudops-infra/ via filing a ticket [here](https://github.com/mozilla-services/github-management/issues/new?assignees=&labels=&template=NewOrgMember.md&title=) similar to [this](https://github.com/mozilla-services/github-management/issues/129).

Reviewer 2: Should we mention [mozilla-services](https://github.com/orgs/mozilla-services/teams/releng/members) and [mozilla-mobile](https://github.com/orgs/mozilla-mobile/teams/releng/members)?

Author: https://github.com/orgs/mozilla-mobile/teams/releng/members is already mentioned at 4) whereas https://github.com/orgs/mozilla-services/teams/releng/members is theoretically granted via 5 but I think the latter can be stressed out as a separate team. Good point, adjusted.

#### Mailing lists

* You should be added to the release@mozilla.com google group as a new hire. This mailing list is managed by Google groups. Owners of this group will be able to add you. Send a test message to release@m.c to verify that your address has been added/subscribed. Talk to your manager if it is not working.
* You should be added to the releng@mozilla.com google group as a new hire. This mailing list is managed by Google groups. Owners of this group will be able to add you. Send a test message to releng@m.c to verify that your address has been added/subscribed. Talk to your manager if it is not working.

:warning:
    release@m.c can contain security-sensitive information. Do not automatically forward your email to a system that is not under Mozilla's control. Same goes for releng@mozilla.com.

### Calendar

Like mail, we now use [Google calendar](https://www.google.com/calendar/)

You'll want to subscribe to the following public calendars:
* [RelEng MoCo calendar](https://calendar.google.com/calendar/u/0?cid=bW96aWxsYS5jb21fMmQzMjM0MzMzMzM1MzAzNjMxMmQzOTM3MzdAcmVzb3VyY2UuY2FsZW5kYXIuZ29vZ2xlLmNvbQ)

> * [RelEng PTO calendar](https://calendar.google.com/calendar/u/0?cid=Y190cWEzaHQ5bXB0ODN1djJzZDJsYjVsaTVva0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t)

Reviewer 2: Worth mentioning "Public - Release Engineering" and "Releases Scheduling"?

Author: Huh, we no longer use the first one. Or it's been renamed to `RelEng MoCo calendar`. Can you give me a link?

Author: `Release Scheduling` is mentioned as part of Releasduty training but I believe it's a good point to circuit it here. Better duplication when it comes to this. Done.

Author: If there's one and I'm wrong, I'll follow-up with another PR, resolving for now.

* CloudOps Jenkins - File a bug similar to [Bug 1721444](https://bugzilla.mozilla.org/show_bug.cgi?id=1721444) and talk to your manager to get access to Janekins CloudOps to be able to debug. You should already have access to cloudops-infra repo if you've done the Github section above.
Once it's properly configured, you'll be able to access it by making some tweaks documented [here](https://github.com/mozilla-services/cloudops-deployment#accessing-jenkins).

### LDAP groups
* RelEng LDAP flags to access various infrastructure pieces. File a bug similar to [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1732846) and ask your manager for vouching. The flags needed are listed below:

```
cn=active_scm_firefoxci,ou=groups,dc=mozilla
cn=all_scm_firefoxci,ou=groups,dc=mozilla
cn=releng,ou=groups,dc=mozilla
cn=RelEngWiki,ou=groups,dc=mozilla
cn=balrog,ou=groups,dc=mozilla
cn=inventory,ou=groups,dc=mozilla
cn=inventory_build,ou=groups,dc=mozilla
cn=vpn_balrog,ou=groups,dc=mozilla
cn=vpn_cloudops_shipit,ou=groups,dc=mozilla
cn=vpn_releng,ou=groups,dc=mozilla
cn=tooltooleditor,ou=groups,dc=mozilla
cn=scm_firefoxci,ou=groups,dc=mozilla
cn=vpn_releng_loan,ou=groups,dc=mozilla
```

> 

Reviewer 2: ```suggestion
You can view your LDAP groups by accessing the [SSO info page](https://sso.mozilla.com/info)
```
